src/server/stream_family.cc

```diff
@@ -1254,6 +1257,14 @@ OpResult<ClaimInfo> OpClaim(const OpArgs& op_args, string_view key, const ClaimO
 
   StreamMemTracker tracker;
 
+  // Try to get the consumer. If not found, create a new one.
+  auto cname = WrapSds(opts.consumer);
+  streamConsumer* consumer = streamLookupConsumer(cgr_res->cg, cname);
+  if (consumer == nullptr)
+    consumer = StreamCreateConsumer(cgr_res->cg, opts.consumer, now_ms, SCC_DEFAULT);
+  else
+    consumer->seen_time = now_ms;
```

Comment:
I want to suggest making a separate function UpdateOrCreateConsumer

Reply:
sure